### PR TITLE
Continuous Integration & StyleCI configurations

### PR DESCRIPTION
**Pull request by ash123456789**
## Description

Since we are adopting StyleCI as a standard it will be ideal to have a standardised configuration across projects. I have added both a Laravel and general PHP configuration which also includes JavaScript settings since more often than not our projects that use PHP will also use some form of JavaScript.

This may need adjusting based on the outcome of #74 since we are currently voting on it. The configurations provided currently assume including whitespace will win 😄 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds documentation)
- [ ] Breaking change (This is a significant change)

## Checklist

- [x] I have ensured that I have used relative links where appropriate.
- [x] I have proofread my documentation.

### CODE REVIEW

**Review by dextermb — APPROVED** [review]
Looks good 👍

File: continuous-integration/styleci/configurations/general-php.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+preset with risky heuristics enabled. It is also tailored to suit our internal PHP & JavaScript guidelines.
+
+```yaml
+php:
+  preset: psr2
+  
+  risky: true
+
+  enabled:
+    - concat_with_spaces
```
Comment (DivineOmega) [inline continuous-integration/styleci/configurations/general-php.md]: I feel like we should leave this out (as default) until consensus is reached in #74.

File: continuous-integration/styleci/configurations/general-php.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  disabled:
+    - short_array_syntax
```
Comment (DivineOmega) [inline continuous-integration/styleci/configurations/general-php.md]: What is the reasoning behind disabling `short_array_syntax`?

File: continuous-integration/styleci/configurations/general-php.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - phpdoc_summary
+    - length_ordered_imports
+    - phpdoc_no_package
+    - concat_without_spaces
```
Comment (DivineOmega) [inline continuous-integration/styleci/configurations/general-php.md]: I feel like we should leave this out (as default) until consensus is reached in #74.

**Review by Jord-JD — CHANGES_REQUESTED** [review]
Great work.

I've added some possible changes/questions. If these are changed in `general-php.md` they should also be reflected in `laravel.php`.

**Review by jaredkove — DISMISSED** [review]

File: continuous-integration/styleci/configurations/general-php.md (hunk 1 above)
Comment (ash123456789) [reply continuous-integration/styleci/configurations/general-php.md]: > I feel like we should leave this out (as default) until consensus is reached in #74.

I left it in as an assumption that whitespaces will win, I had no plans on merging it in until that issue has been decided on.

File: continuous-integration/styleci/configurations/general-php.md (hunk 3 above)
Comment (ash123456789) [reply continuous-integration/styleci/configurations/general-php.md]: https://github.com/langleyfoxall/technologies/pull/76#discussion_r229276821

File: continuous-integration/styleci/configurations/general-php.md (hunk 1 above)
Comment (DivineOmega) [reply continuous-integration/styleci/configurations/general-php.md]: Awesome. :+1:

**Comment by wheatleyjj** [comment]
Happy that we've finally started getting these done. Wish that someone will get the travis config file added too once this is through.

I agree with Jord that we should agree on the content before merging though.

File: continuous-integration/styleci/configurations/general-php.md (hunk 2 above)
Comment (ash123456789) [reply continuous-integration/styleci/configurations/general-php.md]: We do not have an internal guideline on enforcing the short array syntax and I believe in PSR-2 it doesn't specify it either so there was no need to enforce it. Could be worth having a discussion about though.

**Comment by ash123456789** [comment]
> Happy that we've finally started getting these done. Wish that someone will get the travis config file added too once this is through.
> 
> I agree with Jord that we should agree on the content before merging though.

The travis portion was next on my list 😄

File: continuous-integration/styleci/configurations/general-php.md (hunk 2 above)
Comment (dextermb) [reply continuous-integration/styleci/configurations/general-php.md]: `[]` `>` `array()`

Comment (DivineOmega) [reply continuous-integration/styleci/configurations/general-php.md]: I feel like this is commonly used by everyone, but that's a very valid point. Maybe we need a separate issue just to confirm this?

**Comment by ash123456789** [comment]
We will also need to decide what we are doing with the JavaScript validation as well since we currently have no process in that place for that. In one of our projects we use Prettier and StyleCI instead of one or the other and it would ideal to make a decision surrounding that.

**Comment by DivineOmega** [comment]
Considering we may be switched to alternative solutions for CI, may it be worth closing this PR?

Alternatively, we could finish addressing any outstanding discussion and get it merged in soon.

**Comment by DivineOmega** [comment]
@ash123456789 Makes sense to me. :+1: 

When you get a few free minutes, would you be happy to resolve the remaining requested changes and we can get this merged in?

**Pull request closed without merging**